blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
Well, all software engineering is about using old notion in new senses. Our bricks are also not used for real-world building.

I call this file snippets for the reason that its elements are almost like parametrized snippets I can do with my vim plugin. The name `expressions` seems inappropriate for me, because we use this term for Theano variables all across the code, and not for functions returning them. Unfortunately I can not invent a better name, the only other thing I have in mind is `templates`, but I find the confusion even worse then with `snippets`.